proofs/sectorbuilder/utils.go

```diff
@@ -9,7 +9,12 @@ import (
 
 // AddressToProverID creates a prover id by padding an address hash to 31 bytes
 func AddressToProverID(addr address.Address) [31]byte {
-	hash := addr.Hash()
+	// this code will no longer WAE when ID's or BLS pub keys are added
+	// as they will break the assumption of addresses all being the same length
+	if addr.Protocol() != address.Actor && addr.Protocol() != address.SECP256K1 {
+		panic("cannot create prover hash from new address protocol")
```

Comment:
BLOCKING: This code does not assume addresses are all the same length (or even the desired length). This panic should be removed. If we do need that assurance, the check should be where we do make that assumption.

Reply:
I believe [this line will panic]( https://github.com/filecoin-project/go-filecoin/pull/2186/files/480b2a05b80a56b5e81399fb66aa64cdc622f7e0#diff-ab53d6373501c86e59a75d6590d62e7eR23 ) for Protocol 3 addresses since the length of a [BLS public key is 48 bytes](https://github.com/filecoin-project/go-filecoin/blob/master/bls-signatures/types.go#L10).

Reply:
incase the 'this line will panic' link doesn't load for you (it doesn't for me) look at line 26 below.

Reply:
More context, the current address code always returns a slice of [length 20](https://github.com/filecoin-project/go-filecoin/blob/master/address/constants.go#L8) for `addr.Hash()`, since `addr.Payload()` is replacing that call here, and since a BLS public key can be the value of payload, this breaks the assumptions made by this code.